File: 4. Sorting Algo's/Bubble Sort/BubbleSort.java
import java.util.*;

public class BubbleSort {

    public static void sortArray(int[] arr) {
        int n = arr.length;

        for (int i = 0; i < n - 1; i++) {

            boolean swapped = false;

            for (int j = 0; j < n - 1 - i; j++) {

                if (arr[j] > arr[j + 1]) {

                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;

                    swapped = true;
                }
            }

            if (!swapped) {
                break;
            }
        }
    }

    public static void main(String[] args) {

        int[] arr = {7, 3, 2, 5, 1};

        sortArray(arr);

        System.out.println(Arrays.toString(arr));
    }
}

File: 4. Sorting Algo's/Bubble Sort/BubbleSort.py
def bubbleSort(arr):
    l = len(arr)
    for i in range(l - 1):
        swapped = False
        for j in range(l - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                swapped = True
        if not swapped:
            break        

arr = [7, 3, 2, 5, 1]
bubbleSort(arr)
print(arr)